### ISSUE TITLE
Parse Error from Ignored File

### ISSUE THREAD

**Issue by merlinpatt** [eslint/eslint#10615]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **Node Version:** 8.9.2
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
  "eslintIgnore": ["modules/validate.js"],
  "eslintConfig": {
    "parser": "babel-eslint",
    "parserOptions": {
      "ecmaFeatures": {
        "jsx": true
      }
    },
    "plugins": [
      "babel",
      "meteor",
      "react"
    ],
    "extends": [
      "airbnb",
      "plugin:meteor/recommended",
      "plugin:react/recommended"
    ],
    "env": {
      "browser": true
    },
    "globals": {
      "expect": false
    },
    "rules": {
      "import/no-unresolved": 0,
      "import/no-extraneous-dependencies": 0,
      "import/extensions": 0,
      "jsx-a11y/anchor-is-valid": [
        "error",
        {
          "components": [
            "Link"
          ],
          "specialLink": [
            "to"
          ]
        }
      ],
      "max-len": 0,
      "no-alert": 0,
      "no-console": 0,
      "no-restricted-globals": 0,
      "no-return-assign": [
        "error",
        "except-parens"
      ],
      "no-underscore-dangle": [
        "error",
        {
          "allow": [
            "_id",
            "_ensureIndex",
            "_verifyEmailToken",
            "_resetPasswordToken",
            "_name"
          ]
        }
      ],
      "class-methods-use-this": 0,
      "react/jsx-filename-extension": 0,
      "react/forbid-prop-types": 0
    },
    "settings": {
      "import/resolver": "meteor"
    }
  }
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
// validate.js, the root of the issue
import { Meteor } from 'meteor/meteor';

if (Meteor.isClient) {
  import $ from 'jquery';
  import 'jquery-validation';

  export default (form, options) => $(form).validate(options);
}

// any other file importing validate (may be a different path
import validate from '../validate';
// ...
```

```bash
eslint .
```

**What did you expect to happen?**

I have the `validate.js` file in `eslintIgnore` so it should ignore any errors from the `validate.js` file.

**What actually happened? Please include the actual, raw output from ESLint.**

On any file that imports `validate.js`, I get "parse errors in imported module."

```
/Users/merlin/code/clever-beagle/pup/ui/components/AdminUserProfile/AdminUserProfile.js
  12:22  error  Parse errors in imported module '../../../modules/validate': Line 4: 'import' and 'export' may only appear at the top level
```

**Comment by platinumazure** [comment]
This isn't an issue with core ESLint. I think this is actually an issue with eslint-plugin-import not honoring .eslintignore when looking at files that are imported. ESLint core only looks at one file at a time, but eslint-plugin-import actually follows import paths and parses imported files in its own code (and, apparently, without checking to see if the file in question is supposed to be ignored).

I would say either the rule emitting this lint warning should have an option for imports to ignore, or the rule should account for .eslintignore.

In either case, I don't think this is a problem with core ESLint. You should open an issue in eslint-plugin-import. Thanks!

**Comment by merlinpatt** [comment]
Thank you, I've opened an issue over on that repo. Closing this one.
